spatial-tracking-explainer.md

```diff
@@ -0,0 +1,382 @@
+  xrSession.requestFrameOfReference({ type:'stationary', subtype:'floor-level' })
+  .then((frameOfReference) => {
+    xrFrameOfReference = frameOfReference;
+  })
+  .then(setupWebGLLayer)
+  .then(() => {
+    xrSession.requestAnimationFrame(onDrawFrame);
+  });
+}
+```
+
+#### Eye-level Subtype
+
+The origin of this subtype will be initialized at a position near the user's head at the time of creation. The exact `x`, `y`, `z`, and orientation values will be initialized based on the conventions of the underlying platform for stationary eye-level experiences. Some platforms may initialize these values to the user's exact position/orientation at the time of creation. Other platforms that allow users to reset a common eye-level origin shared across multiple apps may use that origin instead.
+
+```js
+let xrSession = null;
+let xrFrameOfReference = null;
+
+function onSessionStarted(session) {
+  xrSession = session;
+  xrSession.requestFrameOfReference({ type:'stationary' , subtype:'eye-level' })
+  .then((frameOfReference) => {
+    xrFrameOfReference = frameOfReference;
+  })
+  .then(setupWebGLLayer)
+  .then(() => {
+    xrSession.requestAnimationFrame(onDrawFrame);
+  });
+}
+```
+
+#### Position-disabled Subtype
+
+The origin of this subtype will be initialized at a position near the user's head at the time of creation.  `XRDevicePose` objects retrieved with this subtype will have varying orientation values but will always report `x`, `y`, `z` values to be `0`.
+
+```js
+let xrSession = null;
+let xrFrameOfReference = null;
+
+function onSessionStarted(session) {
+  xrSession = session;
+  xrSession.requestFrameOfReference({ type:'stationary', subtype:'position-disabled' })
+  .then((frameOfReference) => {
+    xrFrameOfReference = frameOfReference;
+  })
+  .then(setupWebGLLayer)
+  .then(() => {
+    xrSession.requestAnimationFrame(onDrawFrame);
+  });
+}
+```
+
+## Practical Usage Guidelines
+
+### Inline Sessions
+Inline sessions, by definition, do not require a user gesture or user permission to create, and as a result there must be strong limitations on the pose data that can be reported for privacy and security reasons.  Requests for an `XRBoundedFrameOfReference` or an `XRUnboundedFrameOfReference` will always be rejected on inline sessions.  Requests for `XRStationaryFrameOfReference` may succeed, but may also be rejected if the UA is unable provide any tracking information such as for an inline session on a desktop PC or a 2D browser window in a headset.
+
+Additionally, some inline experiences explicitly want tracking information disabled, such as a furniture viewer that will use pointer data to rotate the furniture.  Under both of these situations, there is no `XRFrameOfReference` needed.  Passing `null` as the `XRFrameOfReference` parameter to `getDevicePose()` or `getInputPose()` will return pose data whose values are based on the identity matrix.  Developers are then free to multiply in whatever transform they choose.
+
```

Comment:
Given the threat vectors that caused the https://www.w3.org/TR/orientation-event/ API to never be approved, I am still not inclined to have access to any XR device tracking, even orientation, without some form of user approval or awareness.  I don't see how providing orientation data is going to be any different than the previous `device-orientation` data, in terms of threats;  if apps can get this without user permission, they can open WebXR on any web page and monitor the device's motion.

Reply:
Yeah, absolutely.  That said, the privacy-and-security repo is still working on guidance for what speedbumps, etc. are needed before providing tracking data (and at what granularity, say for diorama-style inline rendering) to the page.  As such, I've migrated the existing behavior into this new design as is.  One of the bullets in the PR description covers filing an issue to address this once the privacy-and-security document is completed.